### ISSUE TITLE
[REVIEW]: PyVOLCANS: A Python package to flexibly explore similarities and differences between volcanic systems

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3649]
**Submitting author:** @PTierz (<a href="http://orcid.org/0000-0001-8889-9900">Pablo Tierz Lopez</a>)
**Repository:** <a href="https://github.com/BritishGeologicalSurvey/pyvolcans" target ="_blank">https://github.com/BritishGeologicalSurvey/pyvolcans</a>
**Version:** v1.3.2
**Editor:** @elbeejay
**Reviewer:** @jifarquharson, @meghanrjones
**Archive:** <a href="https://doi.org/10.5281/zenodo.5809105" target="_blank">10.5281/zenodo.5809105</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/5148a124aa6c9aecf5936022dd1a2bf3/status.svg)](https://joss.theoj.org/papers/5148a124aa6c9aecf5936022dd1a2bf3)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/5148a124aa6c9aecf5936022dd1a2bf3"><img src="https://joss.theoj.org/papers/5148a124aa6c9aecf5936022dd1a2bf3/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/5148a124aa6c9aecf5936022dd1a2bf3/status.svg)](https://joss.theoj.org/papers/5148a124aa6c9aecf5936022dd1a2bf3)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/BritishGeologicalSurvey/pyvolcans" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@jifarquharson & @meghanrjones, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @elbeejay know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @jifarquharson
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/BritishGeologicalSurvey/pyvolcans">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/BritishGeologicalSurvey/pyvolcans">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@PTierz) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @meghanrjones
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/BritishGeologicalSurvey/pyvolcans">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/BritishGeologicalSurvey/pyvolcans">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@PTierz) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @jifarquharson, @meghanrjones it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
PDF failed to compile for issue #3649 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.08 s (129.7 files/s, 48939.1 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                           8            640           1605           1714
Markdown                         1             35              0             97
YAML                             2              9              0             52
-------------------------------------------------------------------------------
SUM:                            11            684           1605           1863
-------------------------------------------------------------------------------


Statistical information for the repository '943343d3818aa42e4d6b7546' was
gathered on 2021/08/23.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
John A Stevenson                17          2649            186           41.44
PTierz                         159          1897            729           38.39
Tierz Lopez                     10           192             54            3.60
mobiuscreek                     22           405            276            9.95
ptierz                           1             1              1            0.03
vyronc                          30           251            200            6.59

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
John A Stevenson           2560           96.6          7.5               15.27
PTierz                     1158           61.0          3.3               11.23
mobiuscreek                  43           10.6          0.6                0.00
vyronc                      198           78.9          5.7                1.52
```

**Comment by elbeejay** [comment]
@whedon generate pdf from branch JOSS_paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch JOSS_paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03649/joss.03649/10.21105.joss.03649.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03649/joss.03649/10.21105.joss.03649.pdf) :page_facing_up: :point_left:

**Comment by elbeejay** [comment]
Hi @jifarquharson and @meghanrjones, thank you both again for agreeing to review this submission! We are currently asking reviewers to try and complete their reviews in 6 weeks. whedon will automatically send you a reminder after 2 weeks, and I'll be checking in throughout the process to answer any questions. If you find you need more time to review just say so in the thread here.

The JOSS review process is entirely open and transparent, and takes place on GitHub. Review comments should be made as issues in the PyVOLCANS repository, please just link this review issue when doing so to help us keep track of things (paste `https://github.com/openjournals/joss-reviews/issues/3649` into the issue text). 


For reference, links to the JOSS reviewer criteria and checklist are below as well:
- [JOSS Review Criteria](https://joss.readthedocs.io/en/latest/review_criteria.html)
- [JOSS Review Checklist](https://joss.readthedocs.io/en/latest/review_checklist.html)


Feel free to ping me (@elbeejay) if you have any questions or concerns. Thanks again for agreeing to review for JOSS.

**Comment by meghanrjones** [comment]
Great, @elbeejay can you please assign this issue to me as a reviewer? Or can I request whedon actions for this?

**Comment by elbeejay** [comment]
@meghanrjones I can try to have whedon re-send the invite, but I believe you should have received an email when this issue was originally opened.

**Comment by elbeejay** [comment]
@whedon re-invite @meghanrjones as reviewer

**Comment by whedon** [comment]
@meghanrjones already has access.

**Comment by whedon** [comment]
:wave: @jifarquharson, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @meghanrjones, please update us on how your review is going (this is an automated reminder).

**Comment by elbeejay** [comment]
Hello all, looks like @jifarquharson has been opening up some issues and has finished (or nearly finished) their review of this submission. @meghanrjones I wanted to check in to see how things are looking on your end. When do you anticipate being able to review PyVOLCANS? 

Thanks all and thanks to @PTierz and co-authors for both their submission and the revisions they have made thus far in response to review comments.

**Comment by meghanrjones** [comment]
> Hello all, looks like @jifarquharson has been opening up some issues and has finished (or nearly finished) their review of this submission. @meghanrjones I wanted to check in to see how things are looking on your end. When do you anticipate being able to review PyVOLCANS?
> 
> Thanks all and thanks to @PTierz and co-authors for both their submission and the revisions they have made thus far in response to review comments.

Hi @elbeejay, thanks for checking in. I'll provide a review within the next week.

**Comment by PTierz** [comment]
Dear @elbeejay, @jifarquharson and @meghanrjones,

Thank you very much for your messages and for all your work on the review of our manuscript.

@jifarquharson - We are hugely thankful for your very positive comments on `PyVOLCANS` and its potential applications. Regarding the typo, many thanks for pointing it out. I have corrected it in the [paper.md](https://github.com/BritishGeologicalSurvey/pyvolcans/blob/JOSS_paper/paper/paper.md) file, in the `JOSS_paper` branch (changes are highlighted in bold text). Finally, concerning [Issue #4](https://github.com/BritishGeologicalSurvey/pyvolcans/issues/4), we hope we will have closed it by the end of this week. Please do not hesitate to let us know if you require anything else from us.

Very many thanks all once again.

Best regards,
Pablo (on behalf of the co-authors)

**Comment by PTierz** [comment]
Dear @meghanrjones, @jifarquharson and @elbeejay,

I sincerely hope that this message finds you well.

I just wanted to let you know that we have preliminarily addressed [Issue #16](https://github.com/BritishGeologicalSurvey/pyvolcans/issues/16) in a new branch called [`volcano-id-profile`](https://github.com/BritishGeologicalSurvey/pyvolcans/tree/volcano-id-profile).

We still need to add a few improvements before we merge the branch into [`main`](https://github.com/BritishGeologicalSurvey/pyvolcans/tree/main) but please feel free to explore the new functionalities of the code and let us know if they appear to be satisfactory as regards the request in the aforementioned Issue.

We will aim at completing the merge by 15th November 2021.

Very many thanks in advance,
Pablo

**Comment by elbeejay** [comment]
Thanks for the update @PTierz

**Comment by PTierz** [comment]
Dear @meghanrjones, @elbeejay and @jifarquharson,

I hope this message finds you well.

I would like to kindly ask you to have a look at [Merge Request #17](https://github.com/BritishGeologicalSurvey/pyvolcans/pull/17) in our PyVOLCANS repository, and let us know whether you consider that the incorporated changes address [Issue #16](https://github.com/BritishGeologicalSurvey/pyvolcans/issues/16) satisfactorily.

A few changes could still be needed as @volcan01010 and/or @mobiuscreek double-check the merge request, but I think the majority of the work to address the aforementioned Issue has been incorporated into the merge request.

Looking forward to hearing from you regarding these new additions to PyVOLCANS. I sincerely hope that they can help the case of making our manuscript suitable for publication in the Journal of Open Source Software.

Very many thanks for all your work to improve our software.

Best regards,
Pablo

**Comment by elbeejay** [comment]
Hi @PTierz,

Thank you for your efforts to add functionality in this new merge request. 

You previously noted that additional tests would be written:

> Test coverage: we can certainly increase the test coverage and include the CLI component. For now, only some of the trickiest functions have tests

do you expect this work effort to take place in a separate branch / PR? 

And I think a more minor but valid suggestion raised by @meghanrjones was:

> One small comment regarding the title/summary: since there is no API documentation for the Python functions, the fact that users interact with PyVOLCANS via the CLI is essential information for those interested in using the tool. Therefore, this information should be included in the title and/or summary.

I agree that at minimum this should be incorporated in the summary section of the paper, if not in the title itself to make it clear to readers/users that they will be interacting with the program via the command-line as opposed to from Python itself.

Thanks,
Jay

**Comment by PTierz** [comment]
Hi @elbeejay,

Very many thanks for your message.

I think we may add the tests for `pyvolcans.py` in a separate merge request from [the one concerning `volcano-id-profile` branch](https://github.com/BritishGeologicalSurvey/pyvolcans/pull/17). Many thanks.

Regarding the specification of the fact that the users interact with PyVOLCANS via the CLI, we had included that information in the [“Statement of Need” section of our manuscript](https://github.com/BritishGeologicalSurvey/pyvolcans/blob/JOSS_paper/paper/paper.md), where we describe our package from a general to more detailed perspectives. Users can also quickly see that they will interact with PyVOLCANS via the command line in the [README.md file](https://github.com/BritishGeologicalSurvey/pyvolcans#readme), which they encounter when accessing the [PyVOLCANS repository](https://github.com/BritishGeologicalSurvey/pyvolcans). We sincerely hope that these specifications can be satisfactory enough. We would really like to keep the title of the manuscript as it is now: we do not see any inaccurate message in it and we think it is very important that it remains as simple and attractive as possible. Very many thanks in advance for your understanding.

Please also note that we have now added the necessary acknowledgements to @jifarquharson and @meghanrjones, as reviewers of the manuscript, and to yourself, as Associate Editor, in the latest version of our manuscript (please see [commit 0b3e1e7 ](https://github.com/BritishGeologicalSurvey/pyvolcans/commit/0b3e1e7c40ed1ec36a79b505d5c7e8faf0d1d01a)). Thank you very much for all your work and support.

Best regards,
Pablo (on behalf of the co-authors)

**Comment by elbeejay** [comment]
> Hi @elbeejay,
> 
> Very many thanks for your message.
> 
> I think we may add the tests for `pyvolcans.py` in a separate merge request from [the one concerning `volcano-id-profile` branch](https://github.com/BritishGeologicalSurvey/pyvolcans/pull/17). Many thanks.
> 
> Regarding the specification of the fact that the users interact with PyVOLCANS via the CLI, we had included that information in the [“Statement of Need” section of our manuscript](https://github.com/BritishGeologicalSurvey/pyvolcans/blob/JOSS_paper/paper/paper.md), where we describe our package from a general to more detailed perspectives. Users can also quickly see that they will interact with PyVOLCANS via the command line in the [README.md file](https://github.com/BritishGeologicalSurvey/pyvolcans#readme), which they encounter when accessing the [PyVOLCANS repository](https://github.com/BritishGeologicalSurvey/pyvolcans). We sincerely hope that these specifications can be satisfactory enough. We would really like to keep the title of the manuscript as it is now: we do not see any inaccurate message in it and we think it is very important that it remains as simple and attractive as possible. Very many thanks in advance for your understanding.
> 
> Please also note that we have now added the necessary acknowledgements to @jifarquharson and @meghanrjones, as reviewers of the manuscript, and to yourself, as Associate Editor, in the latest version of our manuscript (please see [commit 0b3e1e7 ](https://github.com/BritishGeologicalSurvey/pyvolcans/commit/0b3e1e7c40ed1ec36a79b505d5c7e8faf0d1d01a)). Thank you very much for all your work and support.
> 
> Best regards, Pablo (on behalf of the co-authors)

Thanks Pablo. 

In the interest of our reviewers time, could you ping us all once PR#17 as well as the anticipated PR regarding the additional tests for `pyvolcans.py` have been merged? I think it will make more sense for us to all have another look once you and your collaborators are finished making, reviewing, and merging the various changes.

-Jay

**Comment by PTierz** [comment]
Good morning @elbeejay (@jifarquharson, and @meghanrjones),

Many thanks for your message. Certainly. We will do so as soon as the two merges have been carried out. We expect this to happen by close of play on Friday 26th November.

Hopefully, the changes will be satisfactory for you and the reviewers. I personally think that the code will have significantly improved by the time we have incorporated the two PRs.

Very many thanks again for your suggestions to make `PyVOLCANS` a better product.

All the best,
Pablo

**Comment by PTierz** [comment]
Dear @meghanrjones, @jifarquharson and @elbeejay,

I sincerely hope this message finds you well.

Please let me thank you once again for all your work, feedback and suggestions on how to improve the `PyVOLCANS` code and our submission of it to the Journal of Open Source Software.

I am elated to let you know that the latest requested changes have now been incorporated into a new [`v1.3.0`](https://github.com/BritishGeologicalSurvey/pyvolcans/releases/tag/v1.3.0) version of the software. This new version includes two major groups of changes: (1) the added functionality of `PyVOLCANS` to output the _ID profiles_ (i.e. volcanological data available) for both the target volcano and any set of analogue volcanoes generated (please see [merged branch `volcano-id-profile`](https://github.com/BritishGeologicalSurvey/pyvolcans/pull/17); and (2) a comprehensive set of new tests implemented to cover the CLI component in the [`pyvolcans.py` file](https://github.com/BritishGeologicalSurvey/pyvolcans/blob/main/pyvolcans/pyvolcans.py) of the package (please see [merged branch `test-cli`](https://github.com/BritishGeologicalSurvey/pyvolcans/pull/18)).

You can check the first major addition by running commands like: `pyvolcans Kilauea -v` (which prints the Top 10 analogues to Kīlauea as well as the _ID profile_ for the target volcano into the stdout); `pyvolcans Kilauea -v --output_volcano_data` (which saves a `.json` file containing Kīlauea's  _ID profile_); and `pyvolcans Kilauea -v --output_analogues_data` (which saves a different `.json` file containing the _ID profile_ for all the Top 10 analogues to Kīlauea, the latter considering the default, equal-weighting scheme used).

In terms of test coverage, you can now check that `pytest --cov=pyvolcans test\` should inform that over 85% and 90% of [`pyvolcans.py`](https://github.com/BritishGeologicalSurvey/pyvolcans/blob/main/pyvolcans/pyvolcans.py) and [`pyvolcans_func.py`](https://github.com/BritishGeologicalSurvey/pyvolcans/blob/main/pyvolcans/pyvolcans_func.py), respectively, are now covered by tests included in our `PyVOLCANS` package.

We most sincerely hope that these additions to our software can significantly help the case of it meriting publication on the Journal of Open Source Software.

Please do not hesitate to let us know if you required anything else from us.

Best regards,
Pablo (on behalf of the co-authors)

**Comment by PTierz** [comment]
Dear @meghanrjones, @jifarquharson and @elbeejay,

I just wanted to let you know that we have also updated the main [README file](https://github.com/BritishGeologicalSurvey/pyvolcans#readme) in the PyVOLCANS GitHub repository, as well as the [PyVOLCANS Wiki pages](https://github.com/BritishGeologicalSurvey/pyvolcans/wiki) to reflect all changes incorporated up to [PyVOLCANS v1.3.0](https://github.com/BritishGeologicalSurvey/pyvolcans/releases/tag/v1.3.0). If you had any comments or requests, please let us know.

Looking forward to hearing from you.
Thank you very much in advance for your time.

Best regards,
Pablo (on behalf of the co-authors)

**Comment by meghanrjones** [comment]
Thanks for addressing our comments. I will take a look at the revisions within the next two weeks and edit the original review checklist as needed.

**Comment by PTierz** [comment]
Many thanks, @meghanrjones.
Please let us know if you required anything else from us during your review, OK?
Best regards,
Pablo

**Comment by PTierz** [comment]
Dear @meghanrjones, @jifarquharson and @elbeejay,

I hope this message finds you well.

I just wanted to inform you that I have added two recent, but very relevant, citations to our JOSS manuscript, which further show the future potential and applicability of VOLCANS/`PyVOLCANS` to the field of volcanic hazard assessment (please see commits [04095ff](https://github.com/BritishGeologicalSurvey/pyvolcans/commit/04095fffe725ead13f3721236491804158e2db9c#diff-382635c9325bf3273d195ff1b8a44e5b11afd7d97addeb8863ea35feb98c1a07), [ed88cb6](https://github.com/BritishGeologicalSurvey/pyvolcans/commit/ed88cb632eadf0bdfa433e795a1bd86bb2abe617#diff-382635c9325bf3273d195ff1b8a44e5b11afd7d97addeb8863ea35feb98c1a07)). 

I considered that it would be important for future users of the code, and hopefully for future readers of our manuscript, to know more about the context and application of the presented software tool.

Thank you very much in advance.

Kind regards,
Pablo

**Comment by elbeejay** [comment]
@whedon generate pdf from branch JOSS_paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch JOSS_paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03649/joss.03649/10.21105.joss.03649.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03649/joss.03649/10.21105.joss.03649.pdf) :page_facing_up: :point_left:

**Comment by elbeejay** [comment]
@PTierz just so you are aware, I will be asking you to merge the JOSS_paper branch into the main branch at the end of this process. I haven't checked if there are any merge conflicts but wanted to mention this.

**Comment by elbeejay** [comment]
I will just remind you to merge it in as one of the final steps, there's no need to do so now. Just wanted you to be aware. :+1:

**Comment by elbeejay** [comment]
Hello @meghanrjones and @jifarquharson,

Wanted to reach out to you both about this review.

@meghanrjones - have you had a chance to look over the recent revision to the repository? No problem if not; please just update us with an anticipated timeline.

@jifarquharson - several revisions to the submission have taken place since your review was conducted. Would you be willing to have a second look through these changes? Please give us an estimate as to when you will be able to do this check-up.

Thank you both,
Jay

**Comment by jifarquharson** [comment]
Hi @elbeejay, 

> several revisions to the submission have taken place since your review was conducted. Would you be willing to have a second look through these changes? Please give us an estimate as to when you will be able to do this check-up.

I have been keeping up with these as they come in, and all the changes seem to be useful and proportional. I will re-test `pyVOLCANS` tomorrow and note any additional comments that might be warranted. 

Regards,
Jamie

**Comment by meghanrjones** [comment]
Hi @elbeejay,

I apologize for the delay - I have not had a chance to finish the review yet. I will complete the review tomorrow.

Regards,

Meghan

**Comment by jifarquharson** [comment]
Dear @elbeejay and @PTierz et al.,

I have re-read the article, "PyVOLCANS: A Python package to flexibly explore similarities and differences between volcanic systems," and have tested the additional `pyVOLCANS` functionality (now v1.3.0). In particular, the improved 
`--output_analogues_data` and `--output_volcano_data` options increase the range of applications of the tool. I can foresee immediate use-cases of this updated functionality. That being said, a small issue exists that hinders their full utility (there is no disambiguation of .json filenames, and files can be overwritten without warning).  

Pending a fix to [#20](https://github.com/BritishGeologicalSurvey/pyvolcans/issues/20), I am really happy with the revised version and heartily recommend its publication. The minor amendments to the text are warranted, with the added references further demonstrating the timely nature of this contribution. I'm satisfied that all of the checklist items have been satisfied. 

- [ ] Close [#20](https://github.com/BritishGeologicalSurvey/pyvolcans/issues/20).

**Comment by PTierz** [comment]
Dear @jifarquharson,

Thank you very much for all the time and work that you have dedicated to the review of our `PyVOLCANS` code and manuscript. It is hugely appreciated.

Likewise, very many thanks for your support and positive comments about the tool. We sincerely hope that it can be used by and useful for as many volcano scientists and enthusiasts as possible.

We have just posted a comment on Issue [#20](https://github.com/BritishGeologicalSurvey/pyvolcans/issues/20), explaining how we have addressed it and suggesting some acceptance criteria to be checked before the Issue can be closed.

Thank you so much once again.

Best wishes,
Pablo (on behalf of the co-authors)

**Comment by jifarquharson** [comment]
Happy to report that with [`cca2fd6`](https://github.com/BritishGeologicalSurvey/pyvolcans/commit/cca2fd68e6cb79b55799b9f089e155983820672d) and the update to [`v1.3.1`](https://github.com/BritishGeologicalSurvey/pyvolcans/releases/tag/v1.3.1), the disambiguation/overwriting problem ([#20](https://github.com/BritishGeologicalSurvey/pyvolcans/issues/20)) is fixed.

**Comment by meghanrjones** [comment]
Dear all,

I also re-read the article “PyVOLCANS: A Python package to flexibly explore similarities and differences between volcanic systems” and tested out the functionality added since my initial review using version 1.3.1. I am pleased to report that the changes have satisfactorily addressed my concerns for the remaining checklist items "Substantial scholarly effort” and "Automated tests”. I have edited my review checklist accordingly and recommend publication in JOSS after removing the bold formatting on sections that represent changes in the article text. I appreciate the authors' efforts in making the results from VOLCANS accessible for scholarly research through an open-source tool.

I agree with the authors that the current form of the package will be a useful resource for instructors and researchers. While my recommendation for publication does not require additional changes, I advise that the authors consider adding badges to their readme to facilitate quick evaluation of the build/test status, test coverage, and PyPI project page. I also expect that the package would reach a broader audience if the authors add documentation examples for loading the .JSON and .csv output into commonly used Python objects such as a pandas.DataFrame or provide a complementary, simple Python API.

Sincerely,

Meghan Jones

**Comment by elbeejay** [comment]
Thanks for the quick replies @jifarquharson and @meghanrjones, your comments and review of this work are greatly appreciated.

@PTierz I would like to ask that you finalize the `paper.md` file at this time, as Meghan pointed out there is bold formatting in sections that represent changes that are likely not intended to be bold in the final version. I would also suggest the authors consider the suggestions made by Meghan in the above post, namely:

 > consider adding badges to their readme to facilitate quick evaluation of the build/test status, test coverage, and PyPI project page

and

> add documentation examples for loading the .JSON and .csv output into commonly used Python objects such as a pandas.DataFrame or provide a complementary, simple Python API

While neither of these things is a requirement, I do agree that they would increase the user-friendliness of the package and may help people newer to Python or PyVOLCANS. 

In any case, please notify me when you've finalized the paper so that I can go through and type-check it for any minor formatting or reference problems. 

Thanks,
Jay

**Comment by PTierz** [comment]
Dear @meghanrjones and @elbeejay,

Thank you so very much for your messages and for all the work that you have dedicated to reviewing our `PyVOLCANS` software and manuscript. This has resulted in significant improvements to `PyVOLCANS` and we are greatly thankful for that.

Many thanks for your additional suggestions too. We will commit to start working on them after the Christmas break.

I have now removed the bold text from the [`paper.md`](https://github.com/BritishGeologicalSurvey/pyvolcans/blob/JOSS_paper/paper/paper.md) file (please see commit [`97cbb5b`](https://github.com/BritishGeologicalSurvey/pyvolcans/commit/97cbb5b6f42a1a4d9cd140e8517e5350a7f9b663)).

Please do not hesitate to let me know about any other task that I should complete.

Best wishes,
Pablo (on behalf of the co-authors)

**Comment by elbeejay** [comment]
@whedon generate pdf from branch JOSS_paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch JOSS_paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03649/joss.03649/10.21105.joss.03649.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03649/joss.03649/10.21105.joss.03649.pdf) :page_facing_up: :point_left:

**Comment by elbeejay** [comment]
@whedon check references from branch JOSS_paper

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch JOSS_paper
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- None

MISSING DOIs

- 10.1029/2020gc009219 may be a valid DOI for title: Event trees and epistemic uncertainty in long-term volcanic hazard assessment of rift volcanoes: The example of Aluto (Central Ethiopia)
- 10.1007/s00445-019-1336-3 may be a valid DOI for title: VOLCANS: an objective, structured and reproducible method for identifying sets of analogue volcanoes
- 10.1007/s004450100173 may be a valid DOI for title: Constructing event trees for volcanic crises
- 10.1016/j.jvolgeores.2017.08.003 may be a valid DOI for title: WOVOdat–an online, growing library of worldwide volcanic unrest
- 10.5194/nhess-21-3509-2021 may be a valid DOI for title: A unified probabilistic framework for volcanic hazard and eruption forecasting
- 10.1016/b978-0-12-818082-2.00001-9 may be a valid DOI for title: Some relevant issues in volcanic hazard forecasts and management of volcanic crisis

INVALID DOIs

- None
```

**Comment by elbeejay** [comment]
@PTierz and co-authors, I've opened up a [small PR](https://github.com/BritishGeologicalSurvey/pyvolcans/pull/23) to modify a bit of formatting in the `.bib` file. I also made some suggestions for minor edits to the paper itself. The removal of non-DOI links in particular is something I think is important and will help the paper remain durable in the future. The PR should also help resolve some of the missing DOIs whedon has identified above.

Once you've taken action on the paper and the bibliography, please merge your paper branch into the main branch of the code.

Following that, there are a few items that need to be taken care of:

- [ ] Make a tagged release of the software and report the version tag for the archived code here
- [ ] Archive the reviewed code on Zenodo or a similar service (e.g., figshare, an institutional repository)
- [ ] Check that the archived code (e.g., Zenodo) has the correct metadata. The title should match the paper title, the author list should be correct, and you can include authors' ORCID if you wish
- [ ] Provide the DOI of the archived code here

Thanks, please just reply with the above items once they are complete. Once that is done, I will briefly review the paper once more before referring it to one of the EiCs for publication.

**Comment by PTierz** [comment]
Hi @elbeejay,

Thank you so very much for all your help and support.

I proceeded with merging [your suggested PR](https://github.com/BritishGeologicalSurvey/pyvolcans/pull/23) into the [`JOSS_paper` branch](https://github.com/BritishGeologicalSurvey/pyvolcans/tree/JOSS_paper). I also added a couple of commits to the branch. I will wait for your confirmation that these changes look reasonable before I complete the [PR for the branch](https://github.com/BritishGeologicalSurvey/pyvolcans/pull/19), OK?

Very many thanks in advance.

Best regards,
Pablo

**Comment by elbeejay** [comment]
@whedon generate pdf from branch JOSS_paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch JOSS_paper. Reticulating splines etc...
```

**Comment by elbeejay** [comment]
@whedon check references from branch JOSS_paper

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch JOSS_paper
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1029/2020GC009219 is OK
- 10.1007/s00445-019-1336-3 is OK
- 10.1017/CBO9781316276273 is OK
- 10.1007/s004450100173 is OK
- 10.1016/j.jvolgeores.2017.08.003 is OK
- 10.3389/feart.2014.00028 is OK
- 10.7488/era/1087 is OK
- 10.5479/si.GVP.VOTW4-2013 is OK
- 10.5194/nhess-21-3509-2021 is OK
- 10.1016/B978-0-12-818082-2.00001-9 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03649/joss.03649/10.21105.joss.03649.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03649/joss.03649/10.21105.joss.03649.pdf) :page_facing_up: :point_left:

**Comment by elbeejay** [comment]
> Hi @elbeejay,
> 
> Thank you so very much for all your help and support.
> 
> I proceeded with merging [your suggested PR](https://github.com/BritishGeologicalSurvey/pyvolcans/pull/23) into the [`JOSS_paper` branch](https://github.com/BritishGeologicalSurvey/pyvolcans/tree/JOSS_paper). I also added a couple of commits to the branch. I will wait for your confirmation that these changes look reasonable before I complete the [PR for the branch](https://github.com/BritishGeologicalSurvey/pyvolcans/pull/19), OK?
> 
> Very many thanks in advance.
> 
> Best regards, Pablo

@PTierz this looks good to me. At this time I'd say go ahead and merge the paper into the main branch and then proceed to complete the items on that checklist I provided above. If you notice something or want to make minor changes to the paper feel free to go ahead and do so, I'd just ask that you use the whedon commands (like those above) to regenerate the PDF and check the references to ensure nothing has broken in the paper or bibliography files. 

Let me know if you have any questions as you are tagging or archiving the code. 

Thanks,
Jay

**Comment by PTierz** [comment]
Dear @elbeejay,

Thank you very much once again for all your support and assistance during the entire review process.

I have just merged the `JOSS_paper` branch into `main` and released version [`v1.3.2`](https://github.com/BritishGeologicalSurvey/pyvolcans/releases/tag/v1.3.2).

I have also archived [this version on Zenodo](https://zenodo.org/record/5809105#.Ycxiwdso85k) and obtained the following badge:

[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.5809105.svg)](https://doi.org/10.5281/zenodo.5809105)

If there was anything else required from me, please do not hesitate to let me know.

Very many thanks for everything.

Best wishes,
Pablo

**Comment by elbeejay** [comment]
@whedon set v1.3.2 as version

**Comment by whedon** [comment]
OK. v1.3.2 is the version.

**Comment by elbeejay** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1029/2020GC009219 is OK
- 10.1007/s00445-019-1336-3 is OK
- 10.1017/CBO9781316276273 is OK
- 10.1007/s004450100173 is OK
- 10.1016/j.jvolgeores.2017.08.003 is OK
- 10.3389/feart.2014.00028 is OK
- 10.7488/era/1087 is OK
- 10.5479/si.GVP.VOTW4-2013 is OK
- 10.5194/nhess-21-3509-2021 is OK
- 10.1016/B978-0-12-818082-2.00001-9 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by elbeejay** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03649/joss.03649/10.21105.joss.03649.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03649/joss.03649/10.21105.joss.03649.pdf) :page_facing_up: :point_left:

**Comment by elbeejay** [comment]
@whedon set 10.5281/zenodo.5809105 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5809105" target="_blank">10.5281/zenodo.5809105</a> is the archive.

**Comment by elbeejay** [comment]
Cool, everything looks good to me. Big thanks again to @meghanrjones and @jifarquharson for providing constructive and helpful reviews. 

@PTierz and co-authors: At this point I will recommend PyVOLCANS for publication in JOSS. Next an Editor-in-Chief will review the submission and review thread here to double-check that we haven't missed anything and will either notify you of any minor changes needed or will publish your article in JOSS. Thanks for your cooperation, I hope you all found the process to be helpful.

**Comment by elbeejay** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1029/2020GC009219 is OK
- 10.1007/s00445-019-1336-3 is OK
- 10.1017/CBO9781316276273 is OK
- 10.1007/s004450100173 is OK
- 10.1016/j.jvolgeores.2017.08.003 is OK
- 10.3389/feart.2014.00028 is OK
- 10.7488/era/1087 is OK
- 10.5479/si.GVP.VOTW4-2013 is OK
- 10.5194/nhess-21-3509-2021 is OK
- 10.1016/B978-0-12-818082-2.00001-9 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2855

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2855, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1476864079709020189) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2856
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03649](https://doi.org/10.21105/joss.03649)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
@jifarquharson, @meghanrjones – many thanks for your thoughtful and detailed reviews here and to @elbeejay for editing this submission! JOSS relies upon the volunteer effort of people like you and we simply wouldn't be able to do this without you ✨ 

@PTierz – your paper is now accepted and published in JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03649/status.svg)](https://doi.org/10.21105/joss.03649)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03649">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03649/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03649/status.svg
   :target: https://doi.org/10.21105/joss.03649
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03649/status.svg)](https://doi.org/10.21105/joss.03649)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by PTierz** [comment]
Hi everyone,

Apologies for my late reply.

Very many thanks @arfon for your message, and once again to @jifarquharson and @meghanrjones for their invaluable reviews and comments, to @elbeejay for his never-ending support throughout the entire process, and to all the co-authors (@mobiuscreek, @volcan01010) for all the amazing collaborative work they did to help make `PyVOLCANS` happen.

Best wishes to you all!
Pablo

**Comment by elbeejay** [comment]
@openjournals/dev is it possible to add a sentence to the acknowledgements section even though this has been technically published already? The sentence would read:

> Published with permission of the Executive Director of British Geological Survey (NERC-UKRI).

@PTierz please feel free to add further details as you deem appropriate.

**Comment by elbeejay** [comment]
Pinging folks with the requested alteration test from @PTierz, just to make sure this doesn't get lost in the shuffle. Thanks all.

> Very many thanks for this, @elbeejay, and apologies for not having replied before.
> 
> Dear @openjournals/dev, @openjournals/joss-eics,
> 
> The only additions that I would like to make to the Acknowledgements are the following (in bold text):
> 
> "_The research leading to these results has been mainly supported by the UK National Capability Funding (Innovation Flexible Fund programme) **and by the Global Geological Risk Platform of the British Geological Survey NC-ODA grant NE/R000069/1: Geoscience for Sustainable Futures** […] Moreover, we would like to sincerely thank a number of colleagues with whom we shared very insightful conversations about analogue volcanoes and/or PyVOLCANS: Chris Newhall, Isla Sim-
> mons, Adriano Pimentel, Julia Crummy, Gezahegn Yirgu, Charlotte Vye-Brown, Lara Smale,
> Karen Fontijn, Ben Clarke, Susanna Jenkins, Elly Tennant, Pierre Barbillon, Elaine Spiller,
> Philippa White, Teresa Ubide, Sebastián García, Victoria Olivera, Jeremy Pesicek, Vanesa
> Burgos Delgado, Einat Lev, Jonty Rougier, Willy Aspinall, Paolo Papale, Monse Cascante and
> Thomas Giachetti. **Published with permission of the Executive Director of British Geological Survey (NERC-UKRI).**_"
> 
> Would it be possible to add those two sentences to the Acknowledgements without modifying the publishing date?
> 
> I apologise in advance for having missed the inclusion of these sentences during the page-proofing stage.
> 
> Very many thanks in advance.
> 
> Kind regards,
> Pablo

**Comment by arfon** [comment]
> @openjournals/dev is it possible to add a sentence to the acknowledgements section even though this has been technically published already? The sentence would read:

Yes, it's possible. @PTierz – please make the changes to your `paper.md` file (in the repo) and let me (@arfon) know when you've done this.

**Comment by PTierz** [comment]
Dear @arfon,

Thank you very much for your message and your support.

Please accept my apologies for not having replied to this message considerably earlier. I had to be out of the office 22 Jan-1 Feb, and I am catching up with my emails only from this morning.

I have just added the small amending sentences to the Acknowledgements section via commit [1713702](https://github.com/BritishGeologicalSurvey/pyvolcans/commit/1713702886725e02edda4fb032a75d06a51831f9).

If there is anything else required from me, please do not hesitate to let me know. I sincerely hope that the publishing date/year (2021) can be kept, as this would be very important for me.

Very many thanks once again for your support.

Best wishes,
Pablo

**Comment by arfon** [comment]
@PTierz – the paper is now updated. The new PDF might take ~24 hours to show as updated on the JOSS site.

**Comment by PTierz** [comment]
Thank you very much for your support, @arfon, it is extremely appreciated.

I checked the [DOI link to the article](https://doi.org/10.21105/joss.03649) and it seems that it has been already updated!

Many thanks again.
Best wishes,
Pablo
